debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java

```diff
@@ -39,6 +42,15 @@
                                                                                     PostgresConnection.class.getClassLoader());
     private static Logger LOGGER = LoggerFactory.getLogger(PostgresConnection.class);
 
+    private static final String SQL_NON_ARRAY_TYPES = "SELECT t.oid AS oid, t.typname AS name "
+            + "FROM pg_catalog.pg_type t JOIN pg_catalog.pg_namespace n ON (t.typnamespace = n.oid) "
+            + "WHERE n.nspname != 'pg_toast' AND t.typcategory <> 'A'";
+    private static final String SQL_ARRAY_TYPES = "SELECT t.oid AS oid, t.typname AS name, t.typelem AS element "
+            + "FROM pg_catalog.pg_type t JOIN pg_catalog.pg_namespace n ON (t.typnamespace = n.oid) "
+            + "WHERE n.nspname != 'pg_toast' AND t.typcategory = 'A'";
+
+    private TypeRegistry typeRegistry;
```

Comment:
Is there any specific reason for initializing it lazily? Probably should be `volatile` then, but I'd prefer if we initialized it eagerly and made it `final`.